EFCore.NamingConventions.Test/NameRewritingConventionTest.cs

```diff
@@ -116,13 +116,29 @@ public void Foreign_key_reference_navigation()
             entityType.GetForeignKeys().Single().GetConstraintName());
     }
 
+    [Fact]
+    public void Foreign_key_with_explicit_constraint_name()
+    {
+        var model = BuildModel(b => b.Entity<Blog>().HasMany(p => p.Posts)
+            .WithOne(p => p.Blog).HasForeignKey(p => p.BlogId).HasConstraintName("MY_FK_CONSTRAINT"));
+        var entityType = model.FindEntityType(typeof(Post))!;
+        Assert.Equal("MY_FK_CONSTRAINT", entityType.GetForeignKeys().Single().GetConstraintName());
+    }
+
     [Fact]
     public void Index()
     {
         var entityType = BuildEntityType(b => b.Entity<SampleEntity>().HasIndex(s => s.SomeProperty));
         Assert.Equal("ix_sample_entity_some_property", entityType.GetIndexes().Single().GetDatabaseName());
     }
 
+    [Fact]
+    public void Index_with_explicit_name()
```

Comment:
Does this test actually fail before this change? Conventions in EF can never override explicit user configuration anyway, so I'm not sure if the change below is actually needed… Can you please confirm?

Reply:
> Does this test actually fail before this change? Conventions in EF can never override explicit user configuration anyway, so I'm not sure if the change below is actually needed… Can you please confirm?

Yes, this test was failing before the change.

Regarding the second test for a foreign key - this test was not failing and it's there to detect when things may get broken. When I was investigating it, that was working because of a defect in EF. I don't remember all the details, but the code, which is part of EF, was setting an annotation with explicit index name and not firing the event about annotation changed. So, if one day that defect is fixed, explicit index names will be overridden, and the test will show that.